src/components/step-flow/step-flow.component.tsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+import Icon from "../icon";
+import IconButton from "../icon-button";
+import {
+  StyledStepFlow,
+  StyledStepContent,
+  StyledStepContentText,
+  StyledCategoryText,
+  StyledTitleText,
+  StyledTitleTextContainer,
+  StyledVisibleTitleText,
+  StyledHiddenTitleText,
+  StyledStepLabelAndProgress,
+  StyledStepLabel,
+  StyledProgressIndicatorBar,
+  StyledProgressIndicator,
+} from "./step-flow.style";
+import tagComponent, {
+  TagProps,
+} from "../../__internal__/utils/helpers/tags/tags";
+import { VariantTypes } from "../typography";
+import useLocale from "../../hooks/__internal__/useLocale";
+
+export type Steps = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+export interface StepFlowProps extends MarginProps, TagProps {
+  /** A category for the user journey.  */
+  category?: string;
+  /** Set the variant of the internal 'Typography' component which contains the category.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  categoryTypographyVariant?: VariantTypes;
+  /** The title of the current step.  */
+  title: string;
+  /** Set the variant of the internal 'Typography' component which contains the title.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  titleTypographyVariant?: VariantTypes;
+  /** A ref which will be applied to the DOM element which contains the component's title.
+   * Used to focus on the title, to ensure screen reader user's are aware of title changes,
+   * and allow navigation from the StepFlow component to the contents below.
+   */
+  titleRef: React.RefObject<HTMLDivElement>;
```

Comment:
Comment: lets go with the `useImperativeHandle` approach we used in [dialog](https://github.com/Sage/carbon/blob/9d4252a4ad69274be2eff6bc11d5f0f240ec6ffc/src/components/dialog/dialog.component.tsx#L142) here. That way we can support a `forwardRef` (instead of `titleRef`) and only surface the functionality we want here. It will allow us to remove the `titleTabIndex` prop as well (should just be set to -1 imo)